app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -652,9 +666,11 @@ public void onTabReselected(TabLayout.Tab tab) {
                 Utils.makeUserChooseBarcodeFromList(this, barcodeValuesList, new BarcodeValuesListDisambiguatorCallback() {
                     @Override
                     public void onUserChoseBarcode(BarcodeValues barcodeValues) {
-                        cardId = barcodeValues.content();
-                        barcodeType = barcodeValues.format();
-                        barcodeId = "";
+                        Bundle bundle = new Bundle();
+                        bundle.putString(LoyaltyCard.BUNDLE_LOYALTY_CARD_CARD_ID, barcodeValues.content());
+                        bundle.putString(LoyaltyCard.BUNDLE_LOYALTY_CARD_BARCODE_TYPE, barcodeValues.format());
+                        bundle.putString(LoyaltyCard.BUNDLE_LOYALTY_CARD_BARCODE_ID, "");
+                        tempLoyaltyCard.updateFromBundle(bundle);
```

Comment:
FIXME: This needs to be wrapped so the edit activity correctly sets `hasChanged`